quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/IgnoredArtifacts.java

```diff
@@ -131,9 +129,10 @@ private static Set<String> jdbcDrivers() {
             }
         });
 
-        if (vendorsOfAllDatasources.isEmpty()) {
-            vendorsOfAllDatasources.add(Database.Vendor.H2);
-        }
+        // since the default may not be a known property name, look for it explicitly
+        Configuration.getOptionalValue("quarkus.datasource.db-kind")
+            .flatMap(Database::getVendor)
+            .ifPresent(vendorsOfAllDatasources::add);
```

Comment:
```suggestion
        vendorsOfAllDatasources.add(Configuration.getOptionalValue("quarkus.datasource.db-kind")
                .flatMap(Database::getVendor)
                .orElse(Database.Vendor.H2)
        );
```
@shawkins Wouldn't be better to explicitly add the H2 there if we can't get the vendor for the default datasource? Cause in most cases, it'll default to `dev-file`(default value for the KC `--db` option). If there's some unknown DB vendor, validation will fail.

Reply:
quarkus.datasource.db-kind currently has a default value, so it will always resolve to a value. My next changes will remove this case entirely.